File: mobile/lib/model/expense.dart
import 'package:isar/isar.dart';

part 'expense.g.dart';

@collection
class Expense {
  Id id = Isar.autoIncrement;
  late double amount;
  @Index()
  late DateTime dateTime;
  late String note;

  // final category = IsarLink<Category>();
  late int categoryId;

  Expense({
    required this.amount,
    required this.dateTime,
    required this.categoryId,
    this.note = "",
  });
}

File: api/Domain/Entities/Expense.cs
using Domain.Abstractions.Interfaces;
using Domain.ValueObjects;

namespace Domain.Entities;

public class Expense : IAggregateRoot
{
    public EntityId<Expense> Id { get; private set; }
    public EntityId<User> UserId { get; private set; }
    public int LocalId { get; private set; }
    public float Amount { get; private set; }
    public DateTime DateTime { get; private set; }
    public DateTime? UpdatedAt { get; private set; }
    public string? Note { get; private set; } = default!;
    public EntityId<Category> CategoryId { get; private set; }
    public bool IsDeleted { get; private set; } = false;
    private Expense() { }
    public static Expense Create(float amount, EntityId<User> userId, int localId, EntityId<Category> categoryId, DateTime dateTime, string? note = null)
    {
        return new Expense
        {
            LocalId = localId,
            Amount = amount,
            DateTime = dateTime,
            UserId = userId,
            Note = note,
            CategoryId = categoryId
        };
    }
    public void Update(float? amount, EntityId<Category>? categoryId, DateTime? dateTime, string? note = null)
    {
        Amount = amount ?? Amount;
        CategoryId = categoryId ?? CategoryId;
        DateTime = dateTime ?? DateTime;
        Note = note ?? Note;
        UpdatedAt = DateTime.UtcNow;
    }
}